### PR TITLE
Fix RuntimeInformation.IsOSPlatform for Browser/WASM

### PR DESCRIPTION
**Pull request by akoeplinger**
The native code was still using the previous `WEBASSEMBLY` name instead of `BROWSER` as decided in https://github.com/dotnet/runtime/issues/33328.

### CODE REVIEW

File: eng/native/configuretools.cmake [hunk 1]
```diff
@@ -6,7 +6,7 @@ if (CMAKE_C_COMPILER MATCHES "-?[0-9]+(\.[0-9]+)?$")
   set(CLR_CMAKE_COMPILER_FILE_NAME_VERSION "${CMAKE_MATCH_0}")
 endif()
 
-if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_ARCH_WASM)
+if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_BROWSER)
```
Comment (vargaz) [inline eng/native/configuretools.cmake:9]: Note that a lot of this code is targeting all wasm platforms, not just the browser.

File: src/libraries/Native/Unix/CMakeLists.txt [hunk 2]
```diff
@@ -66,6 +65,7 @@ elseif (CLR_CMAKE_TARGET_ARCH_I386)
     add_definitions(-D_FILE_OFFSET_BITS=64)
 elseif (CLR_CMAKE_TARGET_ARCH_WASM)
     add_definitions(-DTARGET_32BIT=1)
+    add_definitions(-D_WASM_)
```
Comment (jkotas) [inline src/libraries/Native/Unix/CMakeLists.txt:68]: We have stopped using the define names with underscores. Can this just use `TARGET_WASM` ?

File: eng/native/configuretools.cmake (hunk 1 above)
Comment (marek-safar) [reply eng/native/configuretools.cmake:9]: I agree that at this level `CLR_CMAKE_TARGET_WASM` would be better, we care about browser at libraries (higher) level

Comment (akoeplinger) [reply eng/native/configuretools.cmake:9]: I disagree, this is looking up tools from the toolchain. `CLR_CMAKE_TARGET_BROWSER` is an explicit alias for Emscripten which only targets Browsers and provides a CMake toolchain file which already defines these variables so we can skip the lookup there.

Other WebAssembly targets might use other toolchains. I'd prefer to keep it like it is.

File: src/libraries/Native/Unix/CMakeLists.txt (hunk 2 above)
Comment (akoeplinger) [reply src/libraries/Native/Unix/CMakeLists.txt]: Fixed.

File: eng/native/configuretools.cmake (hunk 1 above)
Comment (marek-safar) [reply eng/native/configuretools.cmake:9]: I see, let's keep it this way then

**Review by steveisok — APPROVED** [review]

**Pull request merged**